### PR TITLE
Add new h5netcdf backend phony_dims kwarg

### PR DESCRIPTION
**Pull request by kmuehlbauer**
This adds support for the new `phony_dims` keyword for the h5netcdf backend which is available from `h5netcdf` version 0.8.0.

 - [ ] Closes #xxxx
 - [ ] Tests added
 - [x] Passes `isort -rc . && black . && mypy . && flake8`
 - [x] Fully documented, including `whats-new.rst` for all changes and `api.rst` for new API

### CODE REVIEW

File: xarray/backends/h5netcdf_.py [hunk 1]
```diff
@@ -117,13 +118,16 @@ def open(
...
         invalid_netcdf=None,
+        phony_dims=None,
     ):
         import h5netcdf
 
         if format not in [None, "NETCDF4"]:
             raise ValueError("invalid format for h5netcdf backend")
 
         kwargs = {"invalid_netcdf": invalid_netcdf}
+        if LooseVersion(h5netcdf.__version__) >= LooseVersion("0.8.0"):
```
Comment (dcherian) [inline xarray/backends/h5netcdf_.py]: Let's raise an error otherwise when this condition is not satisfied

File: xarray/backends/h5netcdf_.py [hunk 2]
```diff
@@ -117,13 +118,16 @@ def open(
...
+            kwargs["phony_dims"]: phony_dims
```
Comment (mathause) [inline xarray/backends/h5netcdf_.py]: Should this be
```suggestion
            kwargs["phony_dims"] = phony_dims
```
?

File: xarray/backends/h5netcdf_.py (hunk 1 above)
Comment (kmuehlbauer) [reply xarray/backends/h5netcdf_.py]: @dcherian We can do that, but that means users have to use h5netcf >=0.8.0 (which should be OK, I think). Not sure if we should force them. What's the xarray policy for that, BTW? 

Anyway, the whole thing is implemented in h5netcf to be fully backwards compatible. So it should not interfere with users using older versions of h5netcdf.

I'll add the suggested raise first thing tomorrow, if no one stops me.

File: xarray/backends/h5netcdf_.py (hunk 2 above)
Comment (kmuehlbauer) [reply xarray/backends/h5netcdf_.py]: @mathause thanks for spotting, you're right.

File: xarray/backends/h5netcdf_.py (hunk 1 above)
Comment (dcherian) [reply xarray/backends/h5netcdf_.py]: Sorry I wasn't clear enough. What I meant was if a user passes `phony_dims` we should raise if the `h5netcdf` version is not >= 0.8.0

Comment (kmuehlbauer) [reply xarray/backends/h5netcdf_.py]: @dcherian Ah, OK, that makes absolutely sense. Will do.

**Comment by kmuehlbauer** [comment]
Changes added as suggested in the review comments. Sorry for excessive use of force-pushing.

File: xarray/backends/h5netcdf_.py [hunk 3]
```diff
@@ -117,13 +118,22 @@ def open(
...
 
         if format not in [None, "NETCDF4"]:
             raise ValueError("invalid format for h5netcdf backend")
 
         kwargs = {"invalid_netcdf": invalid_netcdf}
+        if phony_dims is not None:
+            if LooseVersion(h5netcdf.__version__) >= LooseVersion("0.8.0"):
+                kwargs["phony_dims"] = phony_dims
+            else:
+                raise (
```
Comment (dcherian) [inline xarray/backends/h5netcdf_.py]: raise `ValueError`?

Comment (kmuehlbauer) [reply xarray/backends/h5netcdf_.py]: @dcherian Sure. I should take more time and be more careful...

**Comment by kmuehlbauer** [comment]
@dcherian This is ready for another review. Please let me know, if I can be of any further assistance.

**Comment by dcherian** [comment]
LGTM. I'll merge in a couple of days if no one else has any comments.

**Comment by dcherian** [comment]
Thanks @kmuehlbauer

**Pull request merged**